src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,186 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { pathExists } from 'fs-extra';
+import * as path from 'path';
+import * as semver from 'semver';
+import { MessageItem, OpenDialogOptions, workspace } from 'vscode';
+import { UserCancelledError } from 'vscode-azureextensionui';
+import { Platform, TemplateFilter } from "../../constants";
+import { ext } from '../../extensionVariables';
+import { validateFuncCoreToolsInstalled } from '../../funcCoreTools/validateFuncCoreToolsInstalled';
+import { localize } from "../../localize";
+import { getFuncExtensionSetting, updateWorkspaceSetting } from '../../ProjectSettings';
+import { cpUtils } from "../../utils/cpUtils";
+import { funcHostTaskId } from "./IProjectCreator";
+import { ScriptProjectCreatorBase } from './ScriptProjectCreatorBase';
+
+export class PythonProjectCreator extends ScriptProjectCreatorBase {
+    public readonly templateFilter: TemplateFilter = TemplateFilter.Verified;
+    private readonly pythonVenvPathSetting: string = 'python.venvRelativePath';
+    private pythonAlias: string;
+    public getLaunchJson(): {} {
+        // currently the Python extension launches Windows with type: python and MacOS with type: pythonExperimental
+        const launchType: string = process.platform === Platform.Windows ? 'python' : 'pythonExperimental';
+        return {
+            version: '0.2.0',
+            configurations: [
+                {
+                    name: localize('azFunc.attachToJavaScriptFunc', 'Attach to Python Functions'),
+                    type: launchType,
+                    request: 'attach',
+                    port: 9091,
+                    host: 'localhost',
+                    preLaunchTask: funcHostTaskId
+                }
+            ]
+        };
+    }
+
+    public async addNonVSCodeFiles(): Promise<void> {
+        // need a custom message for warning as this is a dependency
+        if (await validateFuncCoreToolsInstalled(true)) {
```

Comment:
I don't think `validateFuncCoreToolsInstalled` will work as it is. For example, the message currently says the following:
> You must have the Azure Functions Core Tools installed to debug your local functions.

Which is not true for Python (since you need it for more things). Also we should make sure the dialog is modal since this blocks project creation.

Reply:
Yeah, my comment was to start a conversation about this issue.  Do we want to change `validateFuncCoreToolsInstalled` to allow custom messages or just strip the message out entirely and let the function that calls it handle it?